### PR TITLE
Fix #9756: JpaLazyDataModel: Regression Issue for #9320

### LINKED ISSUE #9756: JpaLazyDataModel: Regression Issue for #9320
### Describe the bug

#9320 was fixed in 12.0.1, after updating to 12.0.3 the issue is back. I believe the refactoring to fix #9594 removed the original fix.

### Reproducer

Reproducers attached to original issues.

### Expected behavior

See original issues.

### PrimeFaces edition

Elite

### PrimeFaces version

12.0.3

### Theme

All

### JSF implementation

Mojarra

### JSF version

2.3

### Java version

11

### Browser(s)

Chrome latest

### PR DESCRIPTION
**Pull request by FlipWarthog**
Fixes #9756

### CODE REVIEW

**Review by melloware — APPROVED** [review]

**Comment by tandraschko** [comment]
What exception is thrown?
Please post msg and stacktrace
Catching everything and log nothing is a no-go

**Comment by FlipWarthog** [comment]
When testing in primefaces-test with Jetty, the exception is

```
SEVERE: javax.el.ELException: Cannot convert [abc] of type [class java.lang.String] to [class java.lang.Long]
        at org.apache.el.lang.ELSupport.coerceToNumber(ELSupport.java:400)
        at org.apache.el.lang.ELSupport.coerceToNumber(ELSupport.java:379)
        at org.apache.el.lang.ELSupport.coerceToType(ELSupport.java:527)
        at org.apache.el.ExpressionFactoryImpl.coerceToType(ExpressionFactoryImpl.java:40)
        at javax.el.ELContext.convertToType(ELContext.java:325)
        at org.primefaces.model.JpaLazyDataModel.convertToType(JpaLazyDataModel.java:353)
        at org.primefaces.model.JpaLazyDataModel.applyFilters(JpaLazyDataModel.java:166)
        at org.primefaces.model.JpaLazyDataModel.count(JpaLazyDataModel.java:116)
```

But I do not believe that's the same as the original issue or what I've seen recently running in Wildfly (but I would need to confirm).

**Comment by tandraschko** [comment]
But how should it ever work? Whats the usecase?
I think i need a reproducer

**Comment by FlipWarthog** [comment]
The jpa branch of primefaces-test shows the issue (attached standalone .zip here). If you run that and try to filter Number Col with a non-numeric value, like "abc", it will throw the exception. This didn't work previously, but the fix for #9320 ([PR here](https://github.com/primefaces/primefaces/pull/9323)) addressed it, but the recent rework removed that fix. 

[PF9756-primefaces-test.zip](https://github.com/primefaces/primefaces/files/10787172/PF9756-primefaces-test.zip)


I just reproduced it in my own application running in Wildfly, and it looks like the Exception being thrown is a NumerFormatException wrapped in an ELException.

File: primefaces/src/main/java/org/primefaces/model/JpaLazyDataModel.java [hunk 1]
```diff
@@ -366,13 +366,18 @@ protected Object convertToType(Object value, Class valueType) {
...
+        try {
+            // first convert the object to string
+            String stringValue = sourceConverter == null
+                    ? value.toString()
+                    : sourceConverter.getAsString(context, UIComponent.getCurrentComponent(context), value);
+
+            // now convert the string to the required target
+            return targetConverter.getAsObject(context, UIComponent.getCurrentComponent(context), stringValue);
+        }
+        catch (Exception e) {
```
Comment (melloware) [inline primefaces/src/main/java/org/primefaces/model/JpaLazyDataModel.java]: So this should probably change to ELException?

**Review by melloware — CHANGES_REQUESTED** [review]

File: primefaces/src/main/java/org/primefaces/model/JpaLazyDataModel.java (hunk 1 above)
Comment (FlipWarthog) [reply primefaces/src/main/java/org/primefaces/model/JpaLazyDataModel.java]: I can change it to ELException. Should I also include IllegalArgumentException, which NumberFormatException extends (among others that could come into play here)?

**Comment by tandraschko** [comment]
i commited something, please try
ELContext must throw a ELException

this does however not solve a ConvertException from a converter, not sure if we should also catch it

**Pull request closed without merging**